domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt

```diff
@@ -71,4 +97,58 @@ class ApplicationLifecycleObserver @Inject constructor(
       }
     }
   }
+
+  private fun logRelativeCpuUsageInBackground(
+    timeIntervalInMillis: Long,
+    currentApplicationState: ApplicationState,
+    currentScreen: ScreenName
+  ) {
+    CoroutineScope(backgroundDispatcher).launch {
+      while (true) {
+        val previousCpuUsageParameters = performanceMetricsController.getLastCpuUsageParameters()
+        val currentCpuUsageParameters = performanceMetricsController.getCurrentCpuUsageParameters(
+          currentApplicationState
+        )
+        val relativeCpuUsage = performanceMetricsController.getRelativeCpuUsage(
+          previousCpuUsageParameters, currentCpuUsageParameters
+        )
+        val applicationState = calculateRelativeApplicationState(
+          previousCpuUsageParameters.applicationState,
+          currentApplicationState
+        )
+        performanceMetricsLogger.logCpuUsage(currentScreen, applicationState, relativeCpuUsage)
+        performanceMetricsController.cacheCpuUsageParameters(currentCpuUsageParameters)
+        delay(timeIntervalInMillis)
+      }
+    }
+  }
+
+  private fun calculateRelativeApplicationState(
+    previousApplicationState: ApplicationState,
+    currentApplicationState: ApplicationState
+  ): ApplicationState {
+    return if (
+      previousApplicationState == ApplicationState.APP_IN_FOREGROUND &&
+      currentApplicationState == ApplicationState.APP_IN_BACKGROUND
+    ) {
+      ApplicationState.FOREGROUND_TO_BACKGROUND
```

Comment:
Why do the transition states matter?

Reply:
In my opinion, transition states will help us gauge the relativeness of the cpu usage value. It'll help us analyse where this value is coming from. For example, we might come across a situation where one cpu value is low and the next is extremely high -- in that case we might be able to judge that the latest cpu value (high) is from a background to foreground switch and the previous one (low) is from a foreground state only.

Reply:
I think one issue is that it may complicate logs to have so many different possible cases. Instead, could we maybe log a "previous screen" value to provide the context for the previous screen (potentially background)? That would allow us to get the same thing, I think, and potentially more (if we populate all foreground screens then it also gives us cross-screen transitions).